ide/db/src/org/netbeans/modules/db/explorer/node/ProcedureNode.java

```diff
@@ -336,39 +337,48 @@ public String getSource() {
             String source = "";
             try {
                 String query = "";
-                String escapedName = "";
+                String escapedName = getName().replace("'", "''");
                 boolean function = false;
                 switch (getType()) {
                     case Function:
                         function = true;
                     case Procedure:
-                        escapedName = getName().replace("'", "''");
-                        query = "SELECT param_list, returns, body, db FROM mysql.proc WHERE name = '"
-                                + escapedName + "';"; // NOI18N
-                        try (Statement stat = connection.getJDBCConnection().createStatement();
-                                ResultSet rs = stat.executeQuery(query);) {
+                        query = "SELECT routine_schema,routine_definition,dtd_identifier,is_deterministic,sql_data_access,routine_comment," // NOI18N
+                              + "(SELECT GROUP_CONCAT(CONCAT(" + (function ? "" : "parameter_mode,' ',") + "parameter_name,' ',dtd_identifier))" // NOI18N
+                              + " FROM information_schema.parameters" // NOI18N
+                              + " WHERE specific_name=routine_name AND ordinal_position>0) AS routine_params" // NOI18N
```

Comment:
This subselect needs an order by, else the DB server is free to reorder.

Reply:
The ordering is needed for the subquery to `information_schema.parameters`. I suspect, that the DB server will always return parameters in the natural order, but if it is not guaranteed an order by will give the safety.